events/equeue/equeue_mbed.cpp

```diff
@@ -131,7 +131,9 @@ static void equeue_sema_timeout(equeue_sema_t *s) {
 bool equeue_sema_wait(equeue_sema_t *s, int ms) {
     int signal = 0;
     Timeout timeout;
-    if (ms > 0) {
+    if (ms == 0) {
+        return false;
+    } else if (ms >= 0) {
```

Comment:
Sounds reasonable.

Throughout the equeue library a timeout of `-1` is used to represent an infinite wait, so here any negative value would indicate that the semaphore waits forever.

I added a bit of clarification to the function declaration since it's easy to miss that corner case at this level: https://github.com/ARMmbed/mbed-os/pull/3991/files#diff-224d7ae23d9d5591b6a59537012c82d9